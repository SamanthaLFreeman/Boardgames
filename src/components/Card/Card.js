import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const Card = ({game, toggleFavorite, toggleOwned, ownedGames, favorites}) => {
  const isFavorite = favorites.map(favorite => favorite.id).includes(game.id);
  const isOwned = ownedGames.map(ownedGame => ownedGame.id).includes(game.id)
  const btnFavClass = isFavorite ? "btn-fav-active" : "btn-fav";
  const btnOwnClass = isOwned ? "btn-own-active" : "btn-own";
  return (
    <article className="card">
      <button className={btnFavClass} onClick={() => toggleFavorite(game, isFavorite)}>Fav</button>
      <button className={btnOwnClass} onClick={() => toggleOwned(game, isOwned)}>Own</button>
      <img className="card-img" src={game.image_url} alt={game.name} />
      <h2>{game.name}</h2>
      <Link to={`/card/${game.id}`} ><button>Details</button></Link>
    </article>
  )
}

export const mapStateToProps = state => ({
  favorites: state.favorites,
  ownedGames: state.ownedGames
})

export default connect(mapStateToProps, null)(Card);