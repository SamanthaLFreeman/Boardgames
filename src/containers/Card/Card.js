import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const Card = ({game, toggleFavorite, toggleOwned, ownedGames, favorites}) => {
  const isFavorite = favorites.map(favorite => favorite.id).includes(game.id);
  const isOwned = ownedGames.map(ownedGame => ownedGame.id).includes(game.id);
  const btnFavClass = isFavorite ? "btn-fav-active" : "btn-fav";
  const btnOwnClass = isOwned ? "btn-own-active" : "btn-own";
  return (
    <article className="card">
      <div className="card-top">
        <div className="card-btns">
          <button className={btnFavClass} onClick={() => toggleFavorite(game, isFavorite)}></button>
          <button className={btnOwnClass} onClick={() => toggleOwned(game, isOwned)}></button>
        </div>
        <img className="card-img" src={game.image_url} alt={game.name} />
        <h2 className="card-name">{game.name}</h2>
      </div>
      <div className="bottom-card">
        <Link to={`/card/${game.id}`} ><button className="details-btn">DETAILS</button></Link>
      </div>
    </article>
  )
}

export const mapStateToProps = state => ({
  favorites: state.favorites,
  ownedGames: state.ownedGames
})

export default connect(mapStateToProps, null)(Card);