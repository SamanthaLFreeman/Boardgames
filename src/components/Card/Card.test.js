import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';

describe('Card', () => {
  let wrapper;
  const mockGame = {
    name: "Catan",
    id: 42
  }
  const toggleFavoriteMock = jest.fn();
  const toggleOwnedMock = jest.fn();
  const ownedGamesMock = [
    {
      name: 'Takenoko',
      id: 41
    }
  ]
  const favoritesMock = [
    {
      name: 'Dead of Winter',
      id: 43
    }
  ]

  beforeEach(() => {
    wrapper = shallow(<Card
      game={mockGame}
      toggleFavorite={toggleFavoriteMock}
      toggleOwned={toggleOwnedMock}
      ownedGames={ownedGamesMock}
      favorites={favoritesMock} />)
  });

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });
})