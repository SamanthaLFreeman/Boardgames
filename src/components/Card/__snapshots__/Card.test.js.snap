// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with the data passed through 1`] = `
<article
  className="card"
>
  <button
    className="btn-fav"
    onClick={[Function]}
  >
    Fav
  </button>
  <button
    className="btn-own"
    onClick={[Function]}
  >
    Own
  </button>
  <img
    alt="Catan"
    className="card-img"
  />
  <h2>
    Catan
  </h2>
  <Link
    to="/card/42"
  >
    <button>
      Details
    </button>
  </Link>
</article>
`;
